vignettes/fork-and-merge.Rmd

```diff
@@ -0,0 +1,157 @@
+---
+title: "Fork and Merge a Dataset"
+description: "How to safely edit a live dataset"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Getting Started}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+One of the main benefits of Crunch is that it lets analysts and clients work with the same datasets. Instead of emailing datasets to clients, you can update the live dataset and ensure that they will see see the most up-to-date information. The potential problem with this setup is that it can become difficult to make provisional changes to the dataset without publishing it to the client. Sometimes an analyst wants to investigate or experiment with a dataset without the risk of sending incorrect or confusing information to the end user This is why we implemented a fork-edit-merge workflow for Crunch datasets. 
+
+"Fork" originates in computer version control systems and just means to take a copy of something with the intention of making some changes to the copy, and then incorporating those changes back into the original. To see how this works lets first upload a new dataset to Crunch. 
+
+```{r, message=FALSE}
+library(crunch)
+```
+```{r, results='hide', echo=FALSE, message=FALSE}
+library(httptest)
+start_vignette("fork-and-merge")
+```
+
+```{r message=FALSE, results='hide'}
+ds <- newDataset(SO_survey, "stackoverflow_survey")
+```
```

Comment:
I think it would look cleaner if these blocks were merged (even if we have to do the trick of calling all of the real ones silently, but then displaying `library(crunch); login(); ds <- newDataset`

Also, we should put `login()` there, it should be obvious, but just to be complete.